### DISCUSSION TITLE
fantastic work! about Expansion and Generalization of Training Methods and Training Datasets

### DISCUSSION THREAD

**Discussion by onefish51** [TheMistoAI/MistoLine#10] [Q&A]
This is truly fantastic work! You have indeed achieved a grand unification of ControlNet for any type of line art input. I have personally tested and found that the support for lineart, softedge, and canny is excellent, better than others' training, and the consistency of the lines is truly great. 
<img width="621" alt="image" src="https://github.com/TheMistoAI/MistoLine/assets/21029719/a5babc66-005d-4bea-a8b6-e694f9df85d4">
<img width="512" alt="image" src="https://github.com/TheMistoAI/MistoLine/assets/21029719/d38333d1-6151-485d-9203-0db93e6e81a6">
<img width="687" alt="image" src="https://github.com/TheMistoAI/MistoLine/assets/21029719/0aca5370-d49d-4f55-9654-d9eb2ea3f894">

I am very curious about how you specifically achieved this level of generalization. Additionally, what order of magnitude of data did you use? What is the data distribution like? I would like to try to see if it can be applied to other types of controls.

**Comment by cseander** [comment]
I'm very curious for how you trained this too as it's leagues better than most other controlnets for SDXL, any information regarding training would be much appreciated.

**Comment by Laidawang** [comment]
+1
